cmds/webboot/webboot.go

```diff
@@ -142,15 +146,25 @@ func (d *DownloadOption) exec(uiEvents <-chan ui.Event, network bool, cacheDir s
 	if err != nil {
 		return nil, err
 	}
-
 	var link string
 	if entry.Label() == customLabel {
 		link, err = menu.PromptTextInput("Enter URL:", validURL, uiEvents)
 		if err != nil {
 			return nil, err
 		}
 	} else {
-		distro := supportedDistros[entry.Label()]
+		//Code for after the specific distro has been selected.
+		//Looks up the distro.
+		distro := supportedDistros[entry.Label()] 
+		if (len(distro.mirrors) > 0) {
+			//Make an array of type menu.Entry to store the mirrors of the 
+			//particular distro selected. Then, display the mirror options.
+			entries := make([]menu.Entry, len(distro.mirrors))
+			for i := range entries {
+				entries[i] = &distro.mirrors[i]
+			}
+			entry, err = menu.PromptMenuEntry("Available Mirrors", "Choose an option:", entries, uiEvents)
+		}
```

Comment:
This looks good. It looks like we also need the logic to make use of the mirror.